apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/layout.tsx

```diff
@@ -0,0 +1,37 @@
+"use client";
+import { Separator } from "@/components/ui/separator";
+import { useLoggedInUser } from "@3rdweb-sdk/react/hooks/useLoggedInUser";
+import { useRouter } from "next/navigation";
+import { toast } from "sonner";
+import { useEcosystemList } from "../hooks/use-ecosystem-list";
+import { EcosystemHeader } from "./components/client/ecosystem-header.client";
+
+export default function Layout({
+  children,
+  params,
+}: { children: React.ReactNode; params: { ecosystemId: string } }) {
+  const { isLoggedIn } = useLoggedInUser();
+  const { ecosystems, isFetched } = useEcosystemList();
+  const router = useRouter();
+
+  if (
+    isFetched &&
+    !ecosystems?.find((ecosystem) => ecosystem.id === params.ecosystemId)
+  ) {
+    // No ecosystem found, redirect to ecosystems home page
+    toast.error("Ecosystem not found");
+    router.replace("/dashboard/connect/ecosystem");
```

Comment:
could also then return `null` here for good measure, react never has to render the children if we redirect